File: GenrChromosome.java
package laga;
import java.util.Random;
import processing.core.PApplet;

/**
 * The GenrChromosome class has all the methods to create and initialize different flavors
 * of chromosomes:
 */

public class GenrChromosome {
	
	PApplet myParent;
	//private int[] arrIndex;
	LagaTools lgTools;
	
	public GenrChromosome (PApplet theParent) {
		myParent = theParent;
		lgTools = new LagaTools();
	}
	/**
	 * the method generate a number chromosome composed by random numbers between a 
	 * minimum parameter and a maximum parameter. The method is based on double numbers.
	 *
	 * @param sizeChromosome -> the size of the chromosome
	 * @param minParam 		 -> the smallest value in the chromosome
	 * @param maxParam 		 -> the bigger value in the chromosome
	 * @return double random chromosome
	 * 
	 */
	public double[] NumberChromosome(int sizeChromosome, double minParam, double maxParam) 
	{
		double[] pop = new double[sizeChromosome];
		Random rand = new Random();
		
		for(int i = 0; i < sizeChromosome; ++i)
		{
			pop[i] = minParam + rand.nextDouble() * (maxParam - minParam);
		}
		
		return pop;
	}

	/**
	 * the method generate a number chromosome composed by random numbers between a 
	 * minimum parameter and a maximum parameter.
	 * the method is based on float numbers.
	 *
	 * @param sizeChromosome -> the size of the chromosome
	 * @param minParam 		 -> the smallest value in the chromosome
	 * @param maxParam 		 -> the bigger value in the chromosome
	 * @return float list in the flavor: float[]
	 * 
	 */
	public float[] NumberChromosome(int sizeChromosome, float minParam, float maxParam) 
	{
		float[] pop = new float[sizeChromosome];
		Random rand = new Random();
		
		for(int i = 0; i < sizeChromosome; ++i)
		{
			pop[i] = minParam + rand.nextFloat() * (maxParam - minParam);
		}
		
		return pop;
	}
	
	/**
	 * the method generate a number chromosome composed by random numbers between a 
	 * minimum parameter and a maximum parameter.
	 * the method is based on int numbers.
	 * 
	 * @param sizeChromosome -> the size of the chromosome
	 * @param minParam 		 -> the smallest value in the chromosome
	 * @param maxParam 		 -> the bigger value in the chromosome
	 * @return int list in the flavor: int[]
	 * 
	 */
	public int[] NumberChromosome(int sizeChromosome, int minParam, int maxParam) 
	{
		int[] pop = new int[sizeChromosome];
		Random rand = new Random();
		
		for(int i = 0; i < sizeChromosome; ++i)
		{
			pop[i] =  rand.nextInt((maxParam - minParam ) + 1 ) + minParam;
		}
		
		return pop;
	}

	/**
	 * This method generate a binary chromosome composed by 1s and 0s in random flavor...
	 * 
	 * @param sizeChromosome -> the size of the population
	 * @return Integer list in the flavor: int[]
	 * 
	 */
	public int[] BinaryChromosome(int sizeChromosome) 
	{
		int[] pop = new int[sizeChromosome];
		int binary;
		
		double v;
		Random rand = new Random();
		
		for(int i = 0; i < sizeChromosome; ++i)
		{
			v = rand.nextDouble();
			if(v < 0.5)
			{
				binary = 0;
			}
			else
			{
				binary = 1;
			}
			pop[i] = binary;
		}
		return pop;
	}

	/**
	 * this method generate a character chromosome composed by lower case characters 
	 * and spaces in a random flavor.
	 * composed by singles characters from a to z plus (space)
	 * 
	 * @param sizeChromosome -> the size of the population
	 * @return char list in the flavor: char[]
	 *  
	 */
	public char[] CharChromosome(int sizeChromosome) 
	{
		int index;
		char[] pop = new char[sizeChromosome];
		Random rand = new Random();
		
		for(int i = 0; i < sizeChromosome; ++i)
		{
			if(rand.nextFloat() < 0.8)
			{
				index = rand.nextInt(26) + 97;
			}
			else
			{
				index = 95;
			}

			pop[i] = (char)index;
		}
		return pop;
	}

	/**
	 * the method generate a number chromosome composed by non repeated numbers between 0 and sizeChromosome(not inclusive).
	 * the method is based on integer numbers. this method is designed by combinatorial problems.
	 * 
	 * @param sizeChromosome -> the size of the chromosome
	 * @return int list in the flavor: int[]
	 * 
	 */
	public int[] NumberChromosomeSwap(int sizeChromosome) 
	{
		int[] pop = new int[sizeChromosome];
		
		for(int i = 0; i < sizeChromosome; ++i)
		{
			pop[i] = i;
		}
		pop = lgTools.fisher_yates(pop); 
		return pop;
	}

}

File: Laga/Laga/Laga/GenrChromosome.cs
using System;
using System.Collections.Generic;
//using System.Linq;
using System.Text;

namespace Laga.GeneticAlgorithm
{
    public class GenrChromosome
    {
        private int size;

        /// <summary>
        /// size of the chromosome
        /// </summary>
        public int SizeChrom
        {
            get
            {
                return size;
            }
            set
            {
                size = value;
            }
        }

        private LagaTools lg;
        private Random rnd;

        /// <summary>
        /// Constructor:
        /// </summary>
        /// <param name="Size">The Size of the Chromosome</param>
        public GenrChromosome(int Size)
        {
            rnd = new Random((int)DateTime.Now.Millisecond);
            this.size = Size;
            lg = new LagaTools();
        }
 
        /// <summary>
        /// the method generates a chromosome composed by random doubles
        /// between min and max.
        /// </summary>
        /// <param name="min">The min value in the chromosome</param>
        /// <param name="max">The max value in the chromosome(exclusive upper bound)</param>
        /// <returns></returns>
        public double[] NumberChromosome(double min, double max)
        {
            double[] chr = new double[size];

            for (int i = 0; i < size; i++)
            {
                chr[i] = min + rnd.NextDouble() * (max - min);
            }

            return chr;
        }

        /// <summary>
        /// the method generates a chromosome composed by random floats
        /// between min and max.
        /// </summary>
        /// <param name="min">The min value in the chromosome</param>
        /// <param name="max">The max value in the chromosome(exclusive upper bound)</param>
        /// <returns></returns>
        public float[] NumberChromosome(float min, float max)
        {
            float[] chr = new float[size];

            for (int i = 0; i < size; i++)
            {
                chr[i] = min + rnd.Next() * (max - min);
            }

            return chr;
        }

        /// <summary>
        /// the method generates a chromosome composed by random integers
        /// between min and max.
        /// </summary>
        /// <param name="min">The min value in the chromosome</param>
        /// <param name="max">The max value in the chromosome(exclusive upper bound)</param>
        /// <returns></returns>
        public int[] NumberChromosome(int min, int max)
        {
            int[] ch = new int[size];

            for (int i = 0; i < size; i++)
            {
                ch[i] = rnd.Next(min, max);
            }
            return ch;

        }

        /// <summary>
        /// creates a binary chromosome composed by 1s and 0s;
        /// </summary>
        /// <returns> a random list of 1s and 0s</returns>
        public int[] BinaryChromosome()
        {
            int[] chr = new int[size];
            int binary;

            double v;

            for (int i = 0; i < size; i++)
            {
                v = rnd.NextDouble();
                binary = (v < 0.5) ? 0 : 1;

                chr[i] = binary;
            }

            return chr;
        }

        /// <summary>
        /// creates an random char chromosome composed by characters.
        /// based on this link: http://www.asciitable.com/
        /// </summary>
        /// <param name="start">the start number for the table, inclusive</param>
        /// <param name="end">the end number for the table, inclusive</param>
        /// <returns>returns a random list of characters</returns>
        /// 
        public char[] CharChromosome(int start, int end)
        {
            char[] chr = new char[size];
            
            for (int i = 0; i < size; i++)
                chr[i] = (char)rnd.Next(start, end + 1);

            return chr;
        }

        /// <summary>
        /// the method generate a number chromosome composed by non repeated numbers between start and start + size(not inclusive).
	    /// the method is based on integer numbers. this method is designed by combinatorial problems.
        /// </summary>
        /// <param name="min">the minimum value in the sequence</param>
        /// /// <param name="max">the maximum value in the sequence</param>
        /// <returns>a non repeat random integer list</returns>
        public int[] NumberChromosomeSwap(int min, int max)
        {
            int[] chr = new int[(max - min) + 1];
            int count = 0;
            for (int i = min; i < max + 1; i++)
            {
                chr[count] = i;
                count++;
            }
            return lg.Fisher_Yates(chr);
        }

    }
}
